kafka/integration_test.go

```diff
@@ -1093,6 +1095,417 @@ func (its *IntegrationTestSuite) TestAdminClient_ListAndDescribeConsumerGroups()
+		"Topic description should not have an error")
+	assert.False(topicDesc.IsInternal, "Topic should not be internal")
+	assert.Empty(topicDesc.AuthorizedOperations,
+		"Topic should not have authorized operations")
+
+	assert.Len(topicDesc.Partitions, 2, "Topic should have two partitions")
+	assert.GreaterOrEqual(len(topicDesc.Partitions[0].Replicas), 1,
+		"At least one replica should exist for partition")
+
+	// Tests for Authorized Operations need a broker with SASL authentication.
+	// This may be a different broker than the usual broker, so we create a
+	// new AdminClient.
+	ac = createAdminClientWithSasl(t)
+	defer ac.Close()
+
+	// Create a topic - the broker may be different for SASL, so we need to
+	// ensure that a topic is created.
+	topic = fmt.Sprintf("%s-%d", testconf.Topic, rand.Int())
+	ctx, cancel = context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+	_, err = ac.CreateTopics(ctx, []TopicSpecification{
+		{
+			Topic:         topic,
+			NumPartitions: 2,
+		},
+	})
+	assert.Nil(err, "CreateTopics should not fail")
+
+	// Delete the second topic after the test is done.
+	defer func(ac *AdminClient) {
+		ctx, cancel = context.WithTimeout(context.Background(), 30*time.Second)
+		defer cancel()
+		_, err = ac.DeleteTopics(ctx, []string{topic})
+		assert.Nil(err, "DeleteTopics should not fail")
+	}(ac)
+
+	// 2. DescribeTopics with default ACLs.
+	ctx, cancel = context.WithTimeout(context.Background(), time.Second*30)
+	defer cancel()
+	topicDescResult, err = ac.DescribeTopics(
+		ctx, TopicCollection{Names: []string{topic}},
+		SetAdminRequestTimeout(30*time.Second),
+		SetAdminOptionIncludeAuthorizedOperations(true))
+
+	assert.Nil(err, "DescribeTopics should not fail")
+
+	topicDescs = topicDescResult.TopicDescriptions
+	assert.Len(topicDescs, 1,
+		"Describing one topic should give exactly one result")
+	assert.Equal(topicDescs[0].Topic, topic,
+		"First result topic should match request topic")
+
+	topicDesc = topicDescs[0]
+	assert.Equal(topicDesc.Error.Code(), ErrNoError,
+		"Topic description should not have an error")
+	assert.NotEmpty(topicDesc.AuthorizedOperations,
+		"Topic should have authorized operations")
+	assert.ElementsMatch(topicDesc.AuthorizedOperations, []ACLOperation{
+		ACLOperationRead, ACLOperationWrite, ACLOperationCreate,
+		ACLOperationDelete, ACLOperationAlter, ACLOperationDescribe,
+		ACLOperationDescribeConfigs, ACLOperationAlterConfigs})
+
+	// Create some ACL bindings on the topic.
+	newACLs := ACLBindings{
+		{
+			Type:                ResourceTopic,
+			Name:                topic,
+			ResourcePatternType: ResourcePatternTypeLiteral,
+			Principal:           "User:*",
+			Host:                "*",
+			Operation:           ACLOperationRead,
+			PermissionType:      ACLPermissionTypeAllow,
+		},
+	}
+
+	ctx, cancel = context.WithTimeout(context.Background(), time.Second*30)
+	defer cancel()
+
+	resultCreateACLs, err := ac.CreateACLs(ctx, newACLs,
```

Comment:
Delete these acls as well

Reply:
Good idea, did the same for describe groups also